heat_pump_adoption_modelling/analysis/MCS/plot_mcs.py

```diff
@@ -0,0 +1,549 @@
+
+
+def ashp_archetypes_boxplot(df, variable):
+    """Produce boxplot of ASHP inflation-adjusted installation costs
+    for each archetype (only considering data from 2019 onwards).
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        Dataframe containing "year", "tech_type",
+        "PROPERTY_TYPE", "BUILT_FORM" and "built_post_1950" variables,
+        as well as the variable specified in the "variable" parameter.
+    variable : str
+        Tested with either "inflated_cost" or "capacity".
+
+    Return
+    ----------
+        No objects returned, but figure is saved as ashp_cost_archetype_boxplots.png
+        if variable is "inflated_cost" and ashp_capacity_archetype_boxplots.png if
+        variable is "capacity".
+    """
+
+    ashp_data_list = []
+
+    archetype_dict = generate_archetype_dict(df)
+    # off_scale = 0
+    # total = 0
+
+    for info in archetype_dict.values():
+        filtered_df = df.loc[
+            (df["year"] >= 2019)
+            & (df["tech_type"] == "Air Source Heat Pump")
+            & info["condition"]
+        ]
+        values = list(filtered_df[variable].dropna())
+        ashp_data_list.append(values)
+        # off_scale += sum([value > 30000 for value in values])
+        # total += len(values)
+
+    ashp_data_list = ashp_data_list[::-1]
+
+    fig, ax = plt.subplots()
+
+    ax.boxplot(
+        ashp_data_list,
+        vert=False,
+        flierprops={
+            "marker": "o",
+            "markersize": 3,
+            "markerfacecolor": "black",
+            "alpha": 0.3,
+        },
+    )
+
+    if variable == "inflated_cost":
+        ax.set_xlim(0, 30000)
+        ax.set_xlabel("Inflation-adjusted cost of installation (£2021)")
+        ax.set_ylabel("Property type")
+        ax.set_title("MCS-certified ASHP installation costs by property type (2019-21)")
+    elif variable == "capacity":
+        ax.set_xlim(0, 30)
+        ax.set_xlabel("ASHP capacity")
+        ax.set_ylabel("Property type")
+        ax.set_title("MCS-certified ASHP capacities by property type (2019-21)")
+
+    ax.grid(axis="x", color="0.8")
+
+    # percentage_off_scale = round(off_scale / total * 100, 2)
+
+    # plt.figtext(0.99, 0.01,
+    #     "Only 2019-21 data used due to increased data quality post-2019.\n{}% of points lie outside axis limits.".format(
+    #         percentage_off_scale
+    #     ), horizontalalignment="right", fontsize = "small"
+    # )
+
+    # Add labels to y axis (requires reversing the list as labels are added
+    # from bottom to top)
+    plt.setp(ax, yticklabels=[info["name"] for info in archetype_dict.values()][::-1])
+    plt.tight_layout()
+
+    if variable == "inflated_cost":
+        plt.savefig(PROJECT_DIR / "outputs/figures/ashp_cost_archetype_boxplots.png")
+    elif variable == "capacity":
+        plt.savefig(
+            PROJECT_DIR / "outputs/figures/ashp_capacity_archetype_boxplots.png"
+        )
```

Comment:
If you want it less wordy, try something like this

```suggestion
    version_name = 'cost' if variable == 'inflated_cost' else 'capacity'
    plt.savefig(PROJECT_DIR / "outputs/figures/ashp_{}_archetype_boxplots.png".format(version_name))
```

Reply:
This is really nice, thanks!